### PR TITLE
Hardening day 1

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: src/main/java/org/sonarlint/intellij/actions/SonarAnalyzeAllFilesAction.java [hunk 1]
```diff
@@ -43,6 +43,7 @@
 import org.sonarlint.intellij.util.SonarLintUtils;
 
 public class SonarAnalyzeAllFilesAction extends AbstractSonarAction {
+
```
Comment (jblievremont) [inline src/main/java/org/sonarlint/intellij/actions/SonarAnalyzeAllFilesAction.java:46]: Nitpick: unnecessary diff

File: src/main/java/org/sonarlint/intellij/notifications/OpenProjectSettingsAction.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+import com.intellij.notification.Notification;
+import com.intellij.notification.NotificationAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.options.ShowSettingsUtil;
+import com.intellij.openapi.project.Project;
+import org.jetbrains.annotations.NotNull;
+import org.sonarlint.intellij.config.project.SonarLintProjectConfigurable;
+
+class OpenProjectSettingsAction extends NotificationAction {
```
Comment (jblievremont) [inline src/main/java/org/sonarlint/intellij/notifications/OpenProjectSettingsAction.java:30]: I wonder if it would be worth refactoring this with `OpenGlobalSettingsAction`, the code is very similar and the only differences are the title string and the `Configurable`.

**Review by jblievremont — APPROVED** [review]
LGTM 👍🏻 !

File: src/main/java/org/sonarlint/intellij/actions/SonarAnalyzeAllFilesAction.java (hunk 1 above)
Comment (henryju) [reply src/main/java/org/sonarlint/intellij/actions/SonarAnalyzeAllFilesAction.java]: Fixed

File: src/main/java/org/sonarlint/intellij/notifications/OpenProjectSettingsAction.java (hunk 2 above)
Comment (henryju) [reply src/main/java/org/sonarlint/intellij/notifications/OpenProjectSettingsAction.java]: Fixed
